### ISSUE TITLE
Trouble installing ps from the source on RHEL 7

### ISSUE THREAD

**Issue by wlandau** [r-lib/ps#171]
I am exploring `ps_fs_info()` from #165 (thanks!) and would like to try it out on a RHEL7 machine. Unfortunately, I am having trouble installing https://github.com/r-lib/ps/commit/83ea0c930761355c72f531b83e220a41597f2fa9 from the source.

```r
> remotes::install_github("r-lib/ps")

Using github PAT from envvar GITHUB_PAT
Downloading GitHub repo r-lib/ps@HEAD
── R CMD build ─────────────────────────────────────────────────────────────────────────────────────────
✔  checking for file ‘/tmp/Rtmpv6DEoP/remotes190615c178d6/r-lib-ps-83ea0c9/DESCRIPTION’ (1.3s)
─  preparing ‘ps’:
✔  checking DESCRIPTION meta-information ...
─  cleaning src
─  running ‘cleanup’
─  checking for LF line-endings in source and make files and shell scripts
─  checking for empty or unneeded directories
─  building ‘ps_1.7.7.9000.tar.gz’
   Warning: invalid uid value replaced by that for user 'nobody'
   
Installing package into ‘/CENSORED/.R/4.3.2’
(as ‘lib’ is unspecified)
* installing *source* package ‘ps’ ...
** using staged installation
** libs
using C compiler: ‘gcc (GCC) 12.2.1 20221121 (Red Hat 12.2.1-4)’
gcc -g -O2  -L/opt/rh/devtoolset-12/root/usr/local/lib64 -L/CENSORED/intel/intel-2020/compilers_and_libraries/linux/mkl/lib/intel64 -Wall px.c -o px
gcc -I"/CENSORED/R-4.3.2/lib64/R/include" -DNDEBUG   -I/opt/rh/devtoolset-12/root/usr/local/include   -fpic  -g -O2  -c init.c -o init.o
gcc -I"/CENSORED/R-4.3.2/lib64/R/include" -DNDEBUG   -I/opt/rh/devtoolset-12/root/usr/local/include   -fpic  -g -O2  -c api-common.c -o api-common.o
gcc -I"/CENSORED/R-4.3.2/lib64/R/include" -DNDEBUG   -I/opt/rh/devtoolset-12/root/usr/local/include   -fpic  -g -O2  -c common.c -o common.o
gcc -I"/CENSORED/R-4.3.2/lib64/R/include" -DNDEBUG   -I/opt/rh/devtoolset-12/root/usr/local/include   -fpic  -g -O2  -c extra.c -o extra.o
gcc -I"/CENSORED/R-4.3.2/lib64/R/include" -DNDEBUG   -I/opt/rh/devtoolset-12/root/usr/local/include   -fpic  -g -O2  -c dummy.c -o dummy.o
gcc -I"/CENSORED/R-4.3.2/lib64/R/include" -DNDEBUG   -I/opt/rh/devtoolset-12/root/usr/local/include   -fpic  -g -O2  -c error-codes.c -o error-codes.o
gcc -I"/CENSORED/R-4.3.2/lib64/R/include" -DNDEBUG   -I/opt/rh/devtoolset-12/root/usr/local/include   -fpic  -g -O2  -c cleancall.c -o cleancall.o
gcc -I"/CENSORED/R-4.3.2/lib64/R/include" -DNDEBUG   -I/opt/rh/devtoolset-12/root/usr/local/include   -fpic  -g -O2  -c posix.c -o posix.o
gcc -I"/CENSORED/R-4.3.2/lib64/R/include" -DNDEBUG   -I/opt/rh/devtoolset-12/root/usr/local/include   -fpic  -g -O2  -c api-posix.c -o api-posix.o
gcc -I"/CENSORED/R-4.3.2/lib64/R/include" -DNDEBUG   -I/opt/rh/devtoolset-12/root/usr/local/include   -fpic  -g -O2  -c linux.c -o linux.o
gcc -I"/CENSORED/R-4.3.2/lib64/R/include" -DNDEBUG   -I/opt/rh/devtoolset-12/root/usr/local/include   -fpic  -g -O2  -c api-linux.c -o api-linux.o
api-linux.c: In function ‘psll_wait’:
api-linux.c:1466:26: error: ‘SYS_pidfd_open’ undeclared (first use in this function); did you mean ‘SYS_mq_open’?
 1466 |     int testfd = syscall(SYS_pidfd_open, getpid(), /* flags= */ 0);
      |                          ^~~~~~~~~~~~~~
      |                          SYS_mq_open
api-linux.c:1466:26: note: each undeclared identifier is reported only once for each function it appears in
make: *** [/CENSORED/R-4.3.2/lib64/R/etc/Makeconf:180: api-linux.o] Error 1
ERROR: compilation failed for package ‘ps’
* removing ‘/CENSORED/.R/4.3.2/ps’
Warning message:
In i.p(...) :
  installation of package ‘/tmp/Rtmpv6DEoP/file19061aac4347/ps_1.7.7.9000.tar.gz’ had non-zero exit status
```

Session info:

<details>

```r
R version 4.3.2 (2023-10-31)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Red Hat Enterprise Linux

Matrix products: default
BLAS/LAPACK: /CENSORED/intel/intel-2020/compilers_and_libraries_2020.0.166/linux/mkl/lib/intel64_lin/libmkl_gf_lp64.so;  LAPACK version 3.7.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: CENSORED
tzcode source: system (glibc)

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
 [1] processx_3.8.2    compiler_4.3.2    R6_2.5.1          rprojroot_2.0.4  
 [5] cli_3.6.1         prettyunits_1.2.0 tools_4.3.2       curl_5.1.0       
 [9] crayon_1.5.2      remotes_2.4.2.1   desc_1.4.3        callr_3.7.3      
[13] ps_1.7.5          pkgbuild_1.4.2 
```

</details>

**Comment by gaborcsardi** [comment]
Can you try again?

**Comment by wlandau** [comment]
https://github.com/r-lib/ps/commit/954f5162eadefb3858ba41c67e8b27846e77d38d seems to have worked, thanks!

**Comment by wlandau** [comment]
On that RHEL 7 machine, I am noticing slowness in `ps_fs_info()`. Do you know of a workaround to make it pretty much instantaneous on these old machines?

```r
> system.time(info <- ps_fs_info(fs::file_create(tempfile())))
   user  system elapsed 
  0.061   0.008  30.269 
> info
                              path mount_point                     name type
1 /tmp/Rtmpnr3NbO/file7fcc3e829973        /tmp /dev/mapper/rootvg-tmplv  xfs
  block_size transfer_block_size total_data_blocks free_blocks
1       4096                4096          26211840    26185335
  free_blocks_non_superuser total_nodes free_nodes
1                  26185335    52428800   52426684
                              id owner  type_code subtype_code MANDLOCK NOATIME
1 02, fd, 00, 00, 00, 00, 00, 00    NA 1481003842           NA    FALSE   FALSE
  NODEV NODIRATIME NOEXEC NOSUID RDONLY RELATIME SYNCHRONOUS NOSYMFOLLOW
1 FALSE      FALSE  FALSE  FALSE  FALSE     TRUE       FALSE       FALSE

> system.time(info <- ps_fs_info("DESCRIPTION"))
info
   user  system elapsed 
  0.003   0.001  72.807 
> info
         path mount_point                 name type block_size
1 DESCRIPTION     /lrlhps nfs-ssd-1fs:/ifs/lrl  nfs     524288
  transfer_block_size total_data_blocks free_blocks free_blocks_non_superuser
1              524288        9279933319   766052930                 395218095
   total_nodes  free_nodes                             id owner type_code
1 3.330043e+12 4.35226e+11 00, 00, 00, 00, 00, 00, 00, 00    NA     26985
  subtype_code MANDLOCK NOATIME NODEV NODIRATIME NOEXEC NOSUID RDONLY RELATIME
1           NA    FALSE   FALSE FALSE      FALSE  FALSE   TRUE  FALSE     TRUE
  SYNCHRONOUS NOSYMFOLLOW
1       FALSE       FALSE
```

**Comment by wlandau** [comment]
If not, it would be great to have a way to tell in advance whether the computation is going to be fast.

**Comment by gaborcsardi** [comment]
Probably the free space query is taking long on NFS. We can certainly improve this, but can you please open a new issue? Thanks.

**Comment by wlandau** [comment]
> We can certainly improve this

Thanks!

> can you please open a new issue? 

Certainly: https://github.com/r-lib/ps/issues/175
